src/main/java/org/thoughtcrime/securesms/util/Prefs.java

```diff
@@ -212,13 +214,36 @@ public static void setChatRingtone(Context context, int accountId, int chatId, U
     return result==null? null : Uri.parse(result);
   }
 
+  private static boolean isAllChatmail(Context context) {
+    DcAccounts dcAccounts = DcHelper.getAccounts(context);
+    for (int accountId : dcAccounts.getAll()) {
+      DcContext dcContext = dcAccounts.getAccount(accountId);
+      if (!dcContext.isChatmail()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
   public static boolean reliableService(Context context) {
+    final String key = "pref_reliable_service";
+    boolean value = false;
+    try {
+      value = getBooleanPreference(context, key, false);
+    } catch(Exception e) {}
+    boolean value2;
     try {
-      return getBooleanPreference(context, "pref_reliable_service", false);
+      value2 = getBooleanPreference(context, key, !value);
+    } catch(Exception e) {
+      value2 = !value;
     }
```

Comment:
took me a while to understand, what is ongoing here, esp. as the whole class is super-simple otherwise :)

this is the kind of code, no one understands when reading out of context in one year :)

maybe better start from scratch with a new variable, `pref_reliable_service2` that has defaults to true.

if the user has really disabled it for some reasons in the past, these reasons may be invalid by the upcoming calls which require more reliable notifications.

otherwise, to check if a key was ever set, `contains()` should do the job as well